dstep/translator/Record.d

```diff
@@ -59,9 +61,17 @@ private:
 
     string writeRecord (string name, void delegate (Data context) dg)
     {
+        auto known = cursor.definition in this.record_definitions;
+        if (known)
+            return "";
+
         auto context = new Data;
-        static if (is(typeof(context.isFwdDeclaration) == bool))
-            context.isFwdDeclaration = !cursor.isDefinition;
+        
+        if (cursor.isDefinition)
+            this.record_definitions[cursor] = true;
+        else
+            context.isFwdDeclaration = true;
+
```

Comment:
I am not sure I understand. Lines you have linked get called _from_ `writeRecord` and are responsible for writing the fields (`CXCursor_FieldDecl`). Struct declarations get written from the return value of `writeRecord` (`return context.data`).